Lucas Grab Fix + Small Tweaks

O1yyy opened the pull request:
Gameplay changes done, ready for backroom review 👍

**UPDATED, check Uair change for details

- Grabs fixed, range standardized

- standing 

https://user-images.githubusercontent.com/19886440/230535916-bf0e029f-e447-4757-92aa-882c87abeedf.mp4

- dash
 
https://user-images.githubusercontent.com/19886440/230536558-8ac9712b-b59f-4f81-8594-b40682623a3f.mp4

- pivot

https://user-images.githubusercontent.com/19886440/230536571-54dc0849-5988-451c-b16b-eb31f3c77204.mp4

  - F1-F2's max range picks up aerial opponents, F3 onwards only grab grounded, goes for all three grabs

- Nair Landing hitbox added

  - general getoffme tool, can lead to aerials at low %'s
    - 4.0 damage
    - 361 angle
    - 100 kbg / 40 bkb
    
https://user-images.githubusercontent.com/19886440/230537392-6a361f8d-c47e-4316-a8ff-01f45000700e.mp4

- Nair multihits shaped into a box style hitbox setup, will help with consistency at DJC range
  - small nerf to damage, 2.0 -> 1.4
  - decreased total Nair shield damage to compensate for new Landing hitbox
  - general cleanup to the acmd and effect scripts

https://user-images.githubusercontent.com/19886440/230592507-704f6ec7-1812-4685-838b-6788827f918f.mp4

- Ftilt small buff to range
  - small range buff to the PSI region only, applies to all 3 angles atm
    -still considering down angle tho
  - reason for this change is for the ftilt to slightly reach farther than jab1-2
  - hitbox window semi-buffed 
    - F7-F10 -> F6-F10
    - all this change does is make the body hitbox hit a frame earlier, so the sweetspot still hits exclusively on F7 because the arm hasn't traveled into the XY plane, nothing too special
    - tldr: the body hitbox hits a frame sooner now
  - cleaned up effects to match new range
  
- side  
![imagen](https://user-images.githubusercontent.com/19886440/230538510-041f805c-249d-40a8-807e-87f9da56d30c.png)

- up

![imagen](https://user-images.githubusercontent.com/19886440/230538631-8ad619a1-2bf7-4b17-9253-131adb132590.png)

- down

![imagen](https://user-images.githubusercontent.com/19886440/230538659-e5ee882d-d754-4482-91c5-c5d6cd471e69.png)

- Standardized DACUS window with an Dash Attack animation tweak
  - pulls perfect boost window to beginning of the move (2f window from the start), anywhere afterwards still gets you an average boost
i would post the anim.zip but i dont want people to copy a whole assets folder for a change they wont even notice
  - no changes to max distance
  
https://user-images.githubusercontent.com/19886440/230538888-633317a7-e176-42ad-b35a-316bc8bf4c54.mp4

- Magnet
  - fixed absorbtion effect to not take up the whole screen
  - made it prettier

https://user-images.githubusercontent.com/19886440/230578334-76bae60a-6a93-4d26-89cd-a9643828e630.mp4



- Uair's FAF was actually left with strange formatting that accidentally made it cumbersome to followup with, this shifts the FAF from 39 to its intended value of 28
  - no changes made to hitbox data, should feel miles better to use now
    - while cleaning up Nair's MotionModule calls, we noticed uair had bungus formatting that was actually making it laggier than intended, BIG thank you to jaccob101 for noticing! 🙏
    - * at first glance this change does allow lucas to AC the aerial or DJ before landing, but common lucas neutral + gameplan almost always favors fullhop uairs or djc uairs, so this change is more so quality of life to allow uair to combo in its intended manner
    
- Dtilt's effect script fixed
![image](https://user-images.githubusercontent.com/19886440/230832166-a2fa7e54-f437-458d-bf2e-7bc258fe6197.png)

O1yyy committed:
Small Tweaks

ftilt stuff

O1yyy committed:
Merge remote-tracking branch 'upstream/dev' into lucas-fixups

SuddyN commented:
will review when changelog is complete

O1yyy committed:
Nair ACMD cleanup + Shine GFX touchups

a lil bit of cleanup, plus a small fix to ftilt to do what i described on the other 2 tilts

SuddyN commented on fighters/lucas/src/acmd/aerials.rs:
```diff
@@ -18,22 +18,23 @@ unsafe fn lucas_attack_air_n_game(fighter: &mut L2CAgentBase) {
     }
-    frame(lua_state, 25.0);
+    frame(lua_state, 23.0);
     if is_excute(fighter) {
-        MotionModule::set_rate(boma, 1.0);
         AttackModule::clear_all(boma);
+        MotionModule::set_rate(boma, 3.0);
     }
     frame(lua_state, 26.0);
     if is_excute(fighter) {
         MotionModule::set_rate(boma, 1.0);
```
lets convert these MotionModule::set_rate calls to the standard `FT_MOTION_RATE(fighter, C / (B - A))` format

`MotionModule::set_rate(boma, 3.0);` --> `FT_MOTION_RATE(fighter, 1.0 / (26.0 - 23.0));` for example

SuddyN replied:
also, you shouldn't wrap them in `is_excute`

O1yyy replied:
i actually think using FT_MOTION_RATE for lucas will be unfeasable because he uses an anim double jump, and rn what FT's doing is speeding the whole fighter up as opposed to just the animation, so rn if i cancel the move while in the sped up parts, i just shoot up, which MotionModule call completely doesnt do so

putting the MotionModule calls outside of the if_execute worked though, i can push that now, but if you still REALLY want those numbers in your formula ima need some help cuz i really dont know what to do with some of these raw doubles calc initially added...

O1yyy replied:
alright, i made a new push addressed this, updating body to reflect changes soon

BIG thanks to jaccob101 🙏🙏

SuddyN commented on fighters/lucas/src/acmd/throws.rs:
```diff
@@ -10,22 +10,39 @@ unsafe fn game_catch(fighter: &mut L2CAgentBase) {
         ArticleModule::change_motion(fighter.module_accessor, *FIGHTER_LUCAS_GENERATE_ARTICLE_HIMOHEBI, Hash40::new("catch"), false, 0.0);
         //FT_MOTION_RATE(fighter, 48.0/(69.0-1.0));
     }
+    frame(lua_state, 8.0);
+    if is_excute(fighter) {
+        FT_MOTION_RATE(fighter, 0.5);
+    }
     frame(lua_state, 9.0);
     if is_excute(fighter) {
+        FT_MOTION_RATE(fighter, 1.0);
```
convert motion rating to standard format and be sure not to wrap them in `is_excute`

SuddyN commented on fighters/lucas/src/acmd/throws.rs:
```diff
@@ -10,22 +10,39 @@ unsafe fn game_catch(fighter: &mut L2CAgentBase) {
     }
     frame(lua_state, 18.0);
     if is_excute(fighter) {
+        FT_MOTION_RATE(fighter, 1.0);
```
convert motion rating to standard format and be sure not to wrap them in `is_excute`

SuddyN commented on fighters/lucas/src/acmd/throws.rs:
```diff
@@ -10,22 +10,39 @@ unsafe fn game_catch(fighter: &mut L2CAgentBase) {
     frame(lua_state, 12.0);
     if is_excute(fighter) {
         //CATCH(fighter, 0, Hash40::new("throw"), 3.0, 0.0, 0.0, 0.5, Some(0.0), Some(0.0), Some(-5.0), *FIGHTER_STATUS_KIND_CAPTURE_PULLED, *COLLISION_SITUATION_MASK_GA);
-        CATCH(fighter, 0, Hash40::new("throw"), 3.0, 0.0, 0.0, 0.0, None, None, None, *FIGHTER_STATUS_KIND_CAPTURE_PULLED, *COLLISION_SITUATION_MASK_GA);
+        CATCH(fighter, 0, Hash40::new("top"), 3.5, 0.0, 6.3, 7.4, None, None, None, *FIGHTER_STATUS_KIND_CAPTURE_PULLED, *COLLISION_SITUATION_MASK_GA);
+        CATCH(fighter, 1, Hash40::new("top"), 3.5, 0.0, 6.3, 12.3, None, None, None, *FIGHTER_STATUS_KIND_CAPTURE_PULLED, *COLLISION_SITUATION_MASK_GA);
     }
     game_CaptureCutCommon(fighter);
+    frame(lua_state, 13.0);
+    if is_excute(fighter) {
+        CATCH(fighter, 2, Hash40::new("top"), 3.5, 0.0, 6.3, 17.0, None, None, None, *FIGHTER_STATUS_KIND_CAPTURE_PULLED, *COLLISION_SITUATION_MASK_GA);
+        CATCH(fighter, 3, Hash40::new("top"), 3.5, 0.0, 6.3, 20.0, None, None, None, *FIGHTER_STATUS_KIND_CAPTURE_PULLED, *COLLISION_SITUATION_MASK_GA);
+    }
     frame(lua_state, 14.0);
     if is_excute(fighter) {
-        //CATCH(fighter, 0, Hash40::new("throw"), 2.8, 0.0, 0.0, 0.0, None, None, None, *FIGHTER_STATUS_KIND_CAPTURE_PULLED, *COLLISION_SITUATION_MASK_GA);
+        grab!(fighter, *MA_MSC_CMD_GRAB_CLEAR, 0);
+        grab!(fighter, *MA_MSC_CMD_GRAB_CLEAR, 1); //Clear the first two, and set the 2 tip grabboxes to ignore aerial opponents
+        CATCH(fighter, 2, Hash40::new("top"), 3.5, 0.0, 6.3, 17.0, None, None, None, *FIGHTER_STATUS_KIND_CAPTURE_PULLED, *COLLISION_SITUATION_MASK_G);
+        CATCH(fighter, 3, Hash40::new("top"), 3.5, 0.0, 6.3, 20.0, None, None, None, *FIGHTER_STATUS_KIND_CAPTURE_PULLED, *COLLISION_SITUATION_MASK_G);
+        FT_MOTION_RATE(fighter, 0.5);
```
you could simplify this a lot with something like this (most grabs are capsules anyways):

frame 12: 
```
CATCH(fighter, 0, Hash40::new("top"), 3.5, 0.0, 6.3, 7.4, Some(0.0), Some(6.3), Some(12.3), *FIGHTER_STATUS_KIND_CAPTURE_PULLED, *COLLISION_SITUATION_MASK_GA);
```

frame 13: 
```
CATCH(fighter, 0, Hash40::new("top"), 3.5, 0.0, 6.3, 7.4, Some(0.0), Some(6.3), Some(20.0), *FIGHTER_STATUS_KIND_CAPTURE_PULLED, *COLLISION_SITUATION_MASK_GA);
```

frame 14: 
```
CATCH(fighter, 0, Hash40::new("top"), 3.5, 0.0, 6.3, 17.0, Some(0.0), Some(6.3), Some(20.0), *FIGHTER_STATUS_KIND_CAPTURE_PULLED, *COLLISION_SITUATION_MASK_G);
```

SuddyN replied:
basic idea: instead of writing 4 different grab boxes and disabling 0-1 after a few frames, just stretch one single grab box to fit the same desired bounds, and redefine those bounds each frame. 

If you agree, you can apply this same idea to the other grabs as well

O1yyy replied:
implemented, looks clean to me!
the only real reason i chose spheres was so it could interpolate but meh its prolly fine like this now

https://user-images.githubusercontent.com/19886440/230767510-b330ac96-5661-4093-aafe-5342a46dc8e7.mp4

SuddyN commented on fighters/lucas/src/acmd/throws.rs:
```diff
@@ -71,6 +96,47 @@ unsafe fn game_catchdash(fighter: &mut L2CAgentBase) {
+#[acmd_script( agent = "lucas", script = "game_catchturn" , category = ACMD_GAME , low_priority)]
+unsafe fn game_catchturn (fighter: &mut L2CAgentBase) {
+	let lua_state = fighter.lua_state_agent;
+	let boma = fighter.boma();
+	if is_excute(fighter) {
+		ArticleModule::generate_article(fighter.module_accessor, *FIGHTER_LUCAS_GENERATE_ARTICLE_HIMOHEBI, false, 0);
+        ArticleModule::change_motion(fighter.module_accessor, *FIGHTER_LUCAS_GENERATE_ARTICLE_HIMOHEBI, Hash40::new("catch_turn"), false, 0.0);
+	}
+	frame(lua_state, 14.0);
+	if is_excute(fighter) {
+		GrabModule::set_rebound(fighter.module_accessor, /*CanCatchRebound*/ true);
+	}
+	frame(lua_state, 15.0);
+	if is_excute(fighter) {
+        CATCH(fighter, 0, Hash40::new("top"), 3.5, 0.0, 6.3, -7.5, None, None, None, *FIGHTER_STATUS_KIND_CAPTURE_PULLED, *COLLISION_SITUATION_MASK_GA);
+        CATCH(fighter, 1, Hash40::new("top"), 3.5, 0.0, 6.3, -13.0, None, None, None, *FIGHTER_STATUS_KIND_CAPTURE_PULLED, *COLLISION_SITUATION_MASK_GA);
+	}
+    game_CaptureCutCommon(fighter);
+	frame(lua_state, 16.0);
+	if is_excute(fighter) {
+        CATCH(fighter, 2, Hash40::new("top"), 3.5, 0.0, 6.3, -17.0, None, None, None, *FIGHTER_STATUS_KIND_CAPTURE_PULLED, *COLLISION_SITUATION_MASK_GA);
+        CATCH(fighter, 3, Hash40::new("top"), 3.5, 0.0, 6.3, -20.0, None, None, None, *FIGHTER_STATUS_KIND_CAPTURE_PULLED, *COLLISION_SITUATION_MASK_GA);
+	}
+    frame(lua_state, 17.0);
+	if is_excute(fighter) {
+        grab!(fighter, *MA_MSC_CMD_GRAB_CLEAR, 0);
+        grab!(fighter, *MA_MSC_CMD_GRAB_CLEAR, 1);
+        CATCH(fighter, 2, Hash40::new("top"), 3.5, 0.0, 6.3, -17.0, None, None, None, *FIGHTER_STATUS_KIND_CAPTURE_PULLED, *COLLISION_SITUATION_MASK_G);
+        CATCH(fighter, 3, Hash40::new("top"), 3.5, 0.0, 6.3, -20.0, None, None, None, *FIGHTER_STATUS_KIND_CAPTURE_PULLED, *COLLISION_SITUATION_MASK_G);
+	}
+	frame(lua_state, 21.0);
+	if is_excute(fighter) {
+		grab!(fighter, *MA_MSC_CMD_GRAB_CLEAR_ALL);
+        GrabModule::set_rebound(fighter.module_accessor, false);
+	}
+	frame(lua_state, 75.0);
+	if is_excute(fighter) {
+		ArticleModule::remove_exist(fighter.module_accessor, *FIGHTER_LUCAS_GENERATE_ARTICLE_HIMOHEBI, ArticleOperationTarget(0));
+	}
+}
```
Please correct the indentation of this function

jaccob101 commented:
Really enjoying the ftilt and nair changes

O1yyy committed:
Code Review Fixes

put as many MotionModule calls outside
+ reworked grab boxes to suggested capsule
+ a small edit to bair's effect script

btw the badly indented script... isnt indented badly on my end? hoping this push magically fixes it 🤷

O1yyy committed:
fix indentation with my bare hands

O1yyy committed:
Uair FAF fix + Nair ACMD cleanup

- Uair's FAF was actually left with strange formatting that accidentally made it cumbersome to followup with, this shifts the FAF from 39 to its intended value of 28
- Nair's ACMD's motion rate calls converted to the standard, cleanup complete

O1yyy committed:
Merge branch 'lucas-fixups' of https://github.com/O1yyy/HewDraw-Remix into lucas-fixups

O1yyy committed:
Small logic fix to Dtilt's effect script

thank you calc!!

O1yyy committed:
Merge remote-tracking branch 'upstream/dev' into lucas-fixups

hdr-balance-backroom reviewed (changes requested):
Grabs should be able to hit both air and ground opponents throughout the entire duration

O1yyy committed:
Merge remote-tracking branch 'upstream/dev' into lucas-fixups

O1yyy committed:
backroom feedback - grab touch up

- add GA mask back to grab late hit

O1yyy commented:
> Grabs should be able to hit both air and ground opponents throughout the entire duration

done! if anything else is needed lmk 👍

jobrien97 reviewed (changes requested):
Nair landing hitbox does not need a `sys_crown` effect. Just adds clutter

O1yyy committed:
Merge remote-tracking branch 'upstream/dev' into lucas-fixups

O1yyy committed:
landing nair effect cleanup

- sys_crown shrunk + sped up to match hitlag
- added back vanilla script's landing smoke

O1yyy commented:
> Nair landing hitbox does not need a `sys_crown` effect. Just adds clutter

done, changed it up a bit, lmk if its good

https://github.com/HDR-Development/HewDraw-Remix/assets/19886440/a9fa8092-7485-4019-94d7-f84a64efa91d

https://github.com/HDR-Development/HewDraw-Remix/assets/19886440/1385bee6-0e3e-4c7f-a509-a27b9532cdb7

jobrien97 reviewed (approved):
isa good
